### PR TITLE
Add signal to attach server and cleanup ssh messages to come out of attach package

### PR DESCRIPTION
**Pull request by fdawg4l**
The majority of this code is cleanup.  We had messages defined in (the main package) of tether but were being used in both tether and PL.

### CODE REVIEW

File: portlayer/attach/messages.go [hunk 1]
```diff
@@ -14,17 +14,100 @@
...
 
+import "golang.org/x/crypto/ssh"
+
 // All of the messages passed over the ssh channel/global mux are (or will be)
 // defined here.
 
-const (
-	WindowChangeReq = "window-change"
-)
+type Messager interface {
```
Comment (hmahmood) [inline portlayer/attach/messages.go]: Should this just be `Message`?

Comment (fdawg4l) [inline portlayer/attach/messages.go]: You don't like back-rubs?  :laughing: 

https://golang.org/doc/effective_go.html#interface-names

It's an interface for sending messages.  So messagER seems right to me.

Comment (hmahmood) [inline portlayer/attach/messages.go]: It seems to be just serializing/deserializing messages and storing its type. Are you planning on adding more methods that would actually send/receive messages?

Comment (fdawg4l) [inline portlayer/attach/messages.go]: I just realized messager != massager.  Please ignore my nonfunny joke.

Comment (fdawg4l) [inline portlayer/attach/messages.go]: You're right;  the convention is for single method interfaces.  

Maybe we'll add more. It does serialize/deserialize and store data in messages we pass over ssh.  Actually sending is tricky.  I may add that later.

I don't feel strongly about the name and can change it to `Message`.

Comment (fdawg4l) [inline portlayer/attach/messages.go]: Done.

**Comment by hmahmood** [comment]
LGTM

**Pull request merged**